docs/reference-architectures/power-platform/eventually-consistent-instances.md

```diff
@@ -0,0 +1,45 @@
+
+# Eventual Consistency in Power Platform
+
+Strategically create data using an upsert to allow systems to synchronise data. Guarantee that the consuming application will be able to accept requests when the data is awaiting synchronisation or missing.
+
+This pattern can be useful in situations when:
+- The system that sends reference data is down.
+- The synchronisation of data takes a long time or the process is delayed.
+- Consuming systems have no logic on the creation of the entity being created. 
+
+## Context and problem
+
+In many modern applications it is vital to consider and implement fault tolerance. In certain situations you may synchronise accounts and contacts for example from one instance of power platform to another. Suppose you have two instances of Dynamics 365 and "Instance A" synchronises data to "Instance B", and another system which reads data from "Instance A" and sends a payload with unique identifiers or alternate keys to "Instance B". When "Instance B" does not have the data, the user will receive a bad request as the entity with that record does not exist.
+
+The following examples show the potential journeys for a record submission. 
+
+**Example 1** - Happy path and all working as expected.
+![Example of multiple system synchronisation](./_images/data-dependent-example.png)
+
+**Example 2** - Unhappy path where sync is down or delayed
+![Example of multiple system synchronisation](./_images/data-dependent-example-unhappy.png)
+
+## Solution
+
+Write a plugin / Flow to always Upsert based on the Guid or Alternate key. This can be performed in a number of plugin steps within the plugin lifecycle. When the entity that you are creating is mandatory then using the Pre Validation step is advised. Pre-Validation happens before any database transactions are started so it is the preferred option if the field is mandatory. However in some scenarios a pre creation plugin step will suffice.
+
+![Solution with Plugin](./_images/solution.png)
+
+
+## Issues and considerations
+
+- Consider the impact of any business logic on an entity that is not hydrated yet.
+- Consider when the data synchronisation is controlled by another team.
```

Comment:
why is this an issue? Could you please add more details?

Reply:
updated with side effects